lib/compat/wordpress-6.1/class-wp-theme-json-6-1.php

```diff
@@ -573,6 +579,46 @@ protected function get_block_classes( $style_nodes ) {
 		return $block_rules;
 	}
 
+	/**
+	 * Converts each styles section into a list of rulesets
+	 * to be appended to the stylesheet.
+	 * These rulesets contain all the css variables (custom variables and preset variables).
+	 *
+	 * See glossary at https://developer.mozilla.org/en-US/docs/Web/CSS/Syntax
+	 *
+	 * For each section this creates a new ruleset such as:
+	 *
+	 *     block-selector {
+	 *       --wp--preset--category--slug: value;
+	 *       --wp--custom--variable: value;
+	 *     }
+	 *
+	 * @param array $nodes Nodes with settings.
+	 * @param array $origins List of origins to process.
+	 * @return string The new stylesheet.
+	 */
+	protected function get_css_variables( $nodes, $origins ) {
+		$stylesheet = '';
+		foreach ( $nodes as $metadata ) {
+			if ( null === $metadata['selector'] ) {
+				continue;
+			}
+
+			$selector = $metadata['selector'];
+
+			if ( static::ROOT_BLOCK_SELECTOR === $selector ) {
+				$selector = ':root';
+			}
+
+			$node         = _wp_array_get( $this->theme_json, $metadata['path'], array() );
+			$declarations = array_merge( static::compute_preset_vars( $node, $origins ), static::compute_theme_vars( $node ) );
+
+			$stylesheet .= static::to_ruleset( $selector, $declarations );
```

Comment:
The reason we moved in https://github.com/WordPress/gutenberg/pull/31302 from :root to body selector was to allow things like margin and padding on the root. This PR uses a different selector for the CSS variables and for the styles I guess it is ok to do that. But in that case, I guess we should have a constant close to ROOT_BLOCK_SELECTOR.

Reply:
See also #35840 which goes into detail on this one. `WP_Theme_JSON::ROOT_BLOCK_SELECTOR` is already in core. 

I can't think of any reason why setting CSS custom properties on `:root` would mean that margin and padding on the `body` wouldn't work; the two requirements aren't related, as far as I can see.

Moving the definitions from `body` to `:root` would only raise one concern, as far as I can see: the code which assigns the definitions would have to be on `:root` in the front end and on `.editor-styles-wrapper` (NOT `:root`!) in the editor.